src/evidently/ui/api/projects.py

```diff
@@ -28,18 +29,24 @@
 from evidently.ui.base import Project
 from evidently.ui.base import ProjectManager
 from evidently.ui.dashboards.base import DashboardPanel
+from evidently.ui.errors import NotEnoughPermissions
 from evidently.ui.type_aliases import OrgID
 from evidently.ui.type_aliases import TeamID
 from evidently.ui.type_aliases import UserID
 from evidently.utils import NumpyEncoder
 
 
-@get("/{project_id:uuid}/reports")
-async def list_reports(
+def is_authenticated(connection: ASGIConnection, _: BaseRouteHandler) -> None:
```

Comment:
Defining this guard here instead of the main app file allows for it to be used as a guard directly on the handlers. This is makes it more explicit to the reader which auth is being performed where and avoids the nested router setup and the factory pattern used before to create the routers.